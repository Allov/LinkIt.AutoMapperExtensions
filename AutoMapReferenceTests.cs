using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using ApprovalTests.Reporters;
using AutoMapper;
using NUnit.Framework;
using RC.Testing;

namespace ShowMeAnExampleOfAutomapperFromLinkedSource
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class AutoMapReferenceTests {

        [Test]
        public void Map_DynamicMappingExpression_ShouldGenerateValidMapping() {
            Mapper
                .CreateMap<LinkedSource, ADto>()
                .MapModel<LinkedSource, AModel, ADto>();
            Mapper.CreateMap<AReference, AReferenceDto>();

            Mapper.AssertConfigurationIsValid();

            var source = CreateSource();

            var actual = Mapper.Map<ADto>(source);

            ApprovalsExt.VerifyPublicProperties(actual);
        }

        private static LinkedSource CreateSource()
        {
            return new LinkedSource{
                Model = new AModel{
                    X = "TheX",
                    Y = "TheY",
                    Reference = 32
                },
                Reference = new AReference
                {
                    A="TheA",
                    B="TheB"
                }
            };
        }
    }
}