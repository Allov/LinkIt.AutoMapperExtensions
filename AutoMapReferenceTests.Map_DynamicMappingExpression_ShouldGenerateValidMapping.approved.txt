{
  "x": "TheX",
  "y": "TheY",
  "reference": {
    "a": "TheA",
    "b": "TheB"
  }
}